miio/ptxswitch.py

```diff
@@ -0,0 +1,250 @@
+        "is_on_1",
+        "is_on_2",
+        "is_on_3",
+        "switchname1",
+        "switchname2",
+        "switchname3"],
+}
+
+
+class PtxSwitchStatus:
+    # Container for status of PTX switch.
+
+    def __init__(self, data: Dict[str, Any]) -> None:
+        self.data = data
+
+    def is_on_index(self, index) -> Optional[bool]:
+        # True if switch {index} is on.
+        k = "is_on_{}".format(index)
+        if k in self.data and self.data[k] is not None:
+            return self.data[k]
+        return None
+
+    @property
+    def is_on_1(self) -> Optional[bool]:
+        # True if switch 1 is on.
+        return self.is_on_index(1)
+
+    @property
+    def is_on_2(self) -> Optional[bool]:
+        # True if switch 2 is on.
+        return self.is_on_index(2)
+
+    @property
+    def is_on_3(self) -> Optional[bool]:
+        # True if switch 3 is on.
+        return self.is_on_index(3)
+
+    def switch_name_index(self, index) -> Optional[str]:
+        # Name of the switch button {index}
+        k = "switchname{}".format(index)
+        if k in self.data and self.data[k] is not None:
+            return self.data[k]
+        return None
+
+    @property
+    def switch_name_1(self) -> Optional[str]:
+        # Name of the switch button 1
+        return self.switch_name_index(1)
+
+    @property
+    def switch_name_2(self) -> Optional[str]:
+        # Name of the switch button 2
+        return self.switch_name_index(2)
+
+    @property
+    def switch_name_3(self) -> Optional[str]:
+        # Name of the switch button 3
+        return self.switch_name_index(3)
+
+    def __repr__(self) -> str:
+        s = "<PtxSwitchStatus " \
+            "is_on_1=%s, " \
+            "is_on_2=%s, "\
+            "is_on_3=%s, " \
+            "switch_name_1=%s, " \
+            "switch_name_2=%s, " \
+            "switch_name_3=%s>" % \
+            (self.is_on_1,
+             self.is_on_2,
+             self.is_on_3,
+             self.switch_name_1,
+             self.switch_name_2,
+             self.switch_name_3)
+        return s
+
+    def __json__(self):
+        return self.data
+
+
+class PtxSwitch(Device):
+    # Main class representing the PTX switch
+
+    def __init__(self, ip: str = None, token: str = None, start_id: int = 0,
+                 debug: int = 0, lazy_discover: bool = True,
+                 model: str = MODEL_PTX_TRIPLE_WALL_SWITCH) -> None:
+        super().__init__(ip, token, start_id, debug, lazy_discover)
+
+        if model in AVAILABLE_PROPERTIES:
+            self.model = model
+        else:
+            self.model = MODEL_PTX_TRIPLE_WALL_SWITCH
+
+    @command(
+        default_output=format_output(
+            "",
+            "Switch 1 Status: {result.is_on_1}\n"
+            "Switch 2 Status: {result.is_on_1}\n"
+            "Switch 2 Status: {result.is_on_1}\n"
+            "Switch 1 Name: {result.switch_name_1}\n"
+            "Switch 2 Name: {result.switch_name_1}\n"
```

Comment:
Should be `result.switch_name_2`. Same applies to `is_on_1`.

Reply:
Sorry... now fixed